lib/native-client.js

```diff
@@ -267,7 +267,11 @@ class NativeClient extends EventEmitter {
    * @param {Function} callback - The callback.
    */
   listDatabases(callback) {
-    this.database.admin().command({ listDatabases: 1 }, {}, (error, result) => {
+    this.database.admin().command({
+      listDatabases: 1
+    }, {
+      readPreference: this.model.readPreference
```

Comment:
this also depends on how is used, if is only used by compass for building the UI i'd go with primary preferred always, what would be the consequence of listing dbs from a secondary?